sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -999,6 +1000,113 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("RewriteGroupByCollation rule works for string") {
+    val dataType = StringType(CollationFactory.collationNameToId("UNICODE_CI"))
+    val schema = StructType(Seq(StructField("name", dataType)))
+    val data = Seq(Row("AA"), Row("aa"), Row("BB"))
+    val df = spark.createDataFrame(spark.sparkContext.parallelize(data), schema)
+    df.createOrReplaceTempView("tempTable")
+    // test RewriteGroupByCollation idempotence
+    val dfGroupBy1 = spark.sql("SELECT COUNT(*) FROM tempTable GROUP BY name")
+    val logicalPlan1 = dfGroupBy1.queryExecution.analyzed
+    val newPlan1 = RewriteGroupByCollation(logicalPlan1)
+    val newNewPlan1 = RewriteGroupByCollation(newPlan1)
+    assert(newPlan1 == newNewPlan1)
+    // get the query execution result
```

Comment:
this comment is not super useful :)
we should be adding more detailed comments for many things, but `checkAnswer` is pretty self descriptive :)

Reply:
fixing